shelley/chain-and-ledger/executable-spec/test/Rules/TestPool.hs

```diff
@@ -148,6 +148,34 @@ registeredPoolIsAdded env ssts =
       M.lookup hk (tSt ^. stPools . to (\(StakePools x) -> x))
         === Just (ledgerSlot env)
 
+-- | Check that a `RetirePool` certificate properly retires a stake pool.
+retiredPoolIsRemoved
+  :: [SourceSignalTarget POOL]
+  -> QC.Property
+retiredPoolIsRemoved ssts =
+  QC.conjoin (map check ssts)
+ where
+  check :: SourceSignalTarget POOL
+        -> QC.Property
+  check sst =
+    case signal sst of
+      -- We omit a well-formedness check for `epoch`, because the executable
+      -- spec will throw a PredicateFailure in that case.
+      RetirePool hk _epoch -> wasRemoved hk
+      _ -> QC.property ()
+   where
+    wasRemoved :: KeyHash -> QC.Property
+    wasRemoved hk = QC.conjoin
+      [ QC.counterexample "hk not in stPools"
+          (hk ∈ dom (source sst ^. (stPools . to (\(StakePools x) -> x))))
+      -- mhueschen: his property fails, likely because the spec allows us to
+      -- retire something which is already in the retiring list.
+      -- , QC.counterexample "hk is already in source's retiring"
+      --     (hk ∉ dom (source sst ^. retiring))
```

Comment:
I *thought* that we had a guarantee that this check would pass, but it seems to consistently fail. This indicates to me that we are allowed to retire something when it's still in the retiring list.

Reply:
yep! retiring when you are already retiring is how you change your retirement  date.

Reply:
Ah thanks! I must have misunderstood a previous discussion about this mechanic. I’ve just removed the commented out assertion.